nss/src/client/mod.rs

```diff
@@ -4,29 +4,19 @@ use tokio::net::UnixStream;
 use tonic::transport::{Channel, Endpoint, Uri};
 use tower::service_fn;
 
-use crate::{info, REQUEST_TIMEOUT};
+use crate::{info, CONNECTION_TIMEOUT};
 
 pub mod authd {
     tonic::include_proto!("authd");
 }
 
 /// new_client creates a new client connection to the gRPC server or returns an active one.
 pub async fn new_client() -> Result<NssClient<Channel>, Box<dyn Error>> {
-    // We need to skip NSS lookups performed by dbus through systemd, otherwise
-    // we could end up in a deadlock due to lookups happening while the authd
-    // daemon is starting up.
-    // This variable is set by systemd specifically for dbus.service to avoid a
-    // similar issue with nss-systemd - we can repurpose it for our case.
-    // ref: https://github.com/systemd/systemd/pull/22552
-    if std::env::var("SYSTEMD_NSS_DYNAMIC_BYPASS").is_ok() {
-        return Err("NSS lookup performed through systemd, skipping...".into());
-    }
```

Comment:
Yeah, it's not ideal. The issue is that, when brokers are DBus activated, we would hang anyway because NSS was never able to connect to authd (due to `dbus-daemon` having this environment set). Do you have any suggestions on how to make this better?
In the worst case, we can decrease the timeout wait.